### ISSUE TITLE
Add possibility to dispose of previous value of ReadOnlyReactiveProperty

### ISSUE THREAD

**Issue by eugen-berenstein** [runceel/ReactiveProperty#166]
Thank you for this awesome library. I'm using ReadOnlyReactiveProperty in my WPF applications to set ViewModels for Pages. I want a page to be disposed of automatically when it is changed. An easy way to do this would be to have ReadOnlyReactiveProperty to dispose of the previous value. I have submitted a pull request for this. What do you think? #165

**Comment by runceel** [comment]
@eugen-berenstein 
Thank you for the feedback.
I thought it is useful if it is in ReactiveProperty. Your approach also work, however how about a following approach? 

```csharp
using Reactive.Bindings;
using System;
using System.Reactive.Disposables;
using System.Reactive.Linq;
using System.Reactive.Subjects;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            var source = new Subject<string>();
            var rp = source.Select(x => new Test(x))
                .DisposeOldValue()
                .ToReadOnlyReactiveProperty();

            Console.WriteLine(rp.Value?.Value);
            source.OnNext("1");
            Console.WriteLine(rp.Value?.Value);
            source.OnNext("2");
            Console.WriteLine(rp.Value?.Value);
            source.OnNext("3");
            Console.WriteLine(rp.Value?.Value);

            rp.Dispose();
        }
    }

    class Test : IDisposable
    {
        public Test(string value)
        {
            Value = value;
        }

        public string Value { get; }

        public void Dispose() => Console.WriteLine($"Dispose: {Value}");
    }

    public static class DisposeOldValueExtensions
    {
        public static IObservable<T> DisposeOldValue<T>(this IObservable<T> self)
            where T : IDisposable => Observable.Create<T>(ox =>
            {
                var d = new SerialDisposable();
                return new CompositeDisposable(
                    self.Do(x => d.Disposable = x).Do(ox.OnNext, ox.OnError, ox.OnCompleted).Subscribe(),
                    d);
            });
    }
}
```

The result:

```

1
Dispose: 1
2
Dispose: 2
3
Dispose: 3
```

If you felt it is good, then I would add this to ReactiveProperty. 
I think the method works to all `IObservable<T>`, so I think it is useful.

How do you think?

**Comment by runceel** [comment]
And would you tell me which name is prefer for this?

- `DisposeOldValue`
- `DisposePreviousValue'

**Comment by runceel** [comment]
I have just submitted a PR for this issue. Could you check the PR?

**Comment by eugen-berenstein** [comment]
I like your approach, I wasn't aware of SerialDisposable. Thank you. I would also prefer the name DisposePreviousValue.

**Comment by runceel** [comment]
@eugen-berenstein Thanks! I'll release a new package which has `DisposePreviousValue`.

**Comment by runceel** [comment]
@eugen-berenstein 
I've released v7.2.0. Thank you for your feedback. :)

https://github.com/runceel/ReactiveProperty/releases/tag/v7.2.0
